contracts/src/xYield/xYieldVault.sol

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { ERC4626 } from "@openzeppelin/contracts/token/ERC20/extensions/ERC4626.sol";
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import { Pausable } from "@openzeppelin/contracts/security/Pausable.sol";
+
+contract xYieldVault is ERC4626, Ownable2Step, ReentrancyGuard, Pausable {
+    using Math for uint256;
+
+    address public guardian;
+    IERC4626 public yieldProtocol;
+
+    uint256 public virtualTotalAssets;
+    uint256 public virtualTotalSupply;
+    bool public isActiveChain;
+
+    mapping(uint64 => address) public siblingVaults;
```

Comment:
Is it relevant to use uint64 here as there's no gas saving ?

Reply:
most ethereum clients use uint64 for chainId. i'm just sticking to that convention

Reply:
Okay makes sense